webapp/src/components/react_select_setting.jsx

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ReactSelect from 'react-select';
+import AsyncSelect from 'react-select/async';
+import CreatableSelect from 'react-select/creatable';
+
+import Setting from 'components/setting';
+import {getStyleForReactSelect} from 'utils/styles';
+
+const MAX_NUM_OPTIONS = 100;
+
+export default class ReactSelectSetting extends React.PureComponent {
+    static propTypes = {
+        name: PropTypes.string,
+        onChange: PropTypes.func,
+        theme: PropTypes.object.isRequired,
+        isClearable: PropTypes.bool,
+        options: PropTypes.array.isRequired,
+        value: PropTypes.oneOfType([
+            PropTypes.object,
+            PropTypes.array,
+            PropTypes.string,
+        ]),
+        addValidate: PropTypes.func,
+        removeValidate: PropTypes.func,
+        required: PropTypes.bool,
+        allowUserDefinedValue: PropTypes.bool,
+        limitOptions: PropTypes.bool,
+        resetInvalidOnChange: PropTypes.bool,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {invalid: false};
+    }
+
+    componentDidMount() {
+        if (this.props.addValidate && this.props.name) {
+            this.props.addValidate(this.props.name, this.isValid);
+        }
+    }
+
+    componentWillUnmount() {
+        if (this.props.removeValidate && this.props.name) {
+            this.props.removeValidate(this.props.name);
+        }
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (prevState.invalid && (this.props.value && this.props.value.value) !== (prevProps.value && prevProps.value.value)) {
+            this.setState({invalid: false}); //eslint-disable-line react/no-did-update-set-state
+        }
+    }
+
+    handleChange = (value) => {
+        if (this.props.onChange) {
+            if (Array.isArray(value)) {
+                this.props.onChange(this.props.name, value.map((x) => x.value));
+            } else {
+                const newValue = value ? value.label : null;
+                this.props.onChange(this.props.name, newValue);
```

Comment:
Why are we using `value.label` to pass to `this.props.onChange`? That's certainly not what `ReactSelectSetting` should be doing, and is not what it does in the Jira plugin either. 4/5 We should be using `value.value`.

Reply:
@jfrerich :point_up:

Reply:
@hanzei I'll make this into a ticket. It's not causing any functional issues because the label and value are the same in the case of the GitHub repo options, but it is certainly not how we should be using this component.